libcloudforensics/providers/azure/internal/monitoring.py

```diff
@@ -0,0 +1,135 @@
+# limitations under the License.
+"""Azure Monitoring functionality."""
+
+from typing import List, Optional, Dict, TYPE_CHECKING
+
+# Pylint complains about the import but the library imports just fine,
+# so we can ignore the warning.
+# pylint: disable=import-error
+from azure.mgmt.monitor import MonitorManagementClient
+from azure.mgmt.monitor.v2018_01_01 import models
+# pylint: enable=import-error
+
+from libcloudforensics.providers.azure.internal import account
+
+if TYPE_CHECKING:
+  from datetime import datetime
+
+
+class AZMonitoring:
+  """Azure Monitoring.
+
+  Attributes:
+    az_account (AZAccount): An Azure account object.
+    monitoring_client (MonitorManagementClient): An Azure monitoring client
+        object.
+  """
+
+  def __init__(self,
+               az_account: account.AZAccount) -> None:
+    """Initialize the Azure monitoring class.
+
+    Args:
+      az_account (AZAccount): An Azure account object.
+    """
+    self.az_account = az_account
+    self.monitoring_client = MonitorManagementClient(
+        self.az_account.credentials, self.az_account.subscription_id)
+
+  def ListAvailableMetricsForResource(self, resource_id: str) -> List[str]:
+    """List the available metrics for a given resource.
+
+      Args:
+        resource_id (str): The resource ID from which to list available
+            metrics.
+
+      Returns:
+        List[str]: A list of metrics that can be queried for the resource ID.
+
+      Raises:
+        RuntimeError: If the resource could not be found.
+    """
+    try:
+      return [metric.name.value for metric
+              in self.monitoring_client.metric_definitions.list(resource_id)]
+    except models.ErrorResponseException:
+      raise RuntimeError('Could not fetch metrics for resource {0:s}. Please '
+                         'make sure you specified the full resource ID url, '
+                         'i.e. /subscriptions/<>/resourceGroups/<>/providers'
+                         '/<>/<>/yourResourceName'.format(resource_id))
+
+  def GetMetricsForResource(
+      self,
+      resource_id: str,
+      metrics: str,
+      from_date: Optional['datetime'] = None,
+      to_date: Optional['datetime'] = None,
+      interval: Optional[str] = None,
+      aggregation: str = 'Total',
+      qfilter: Optional[str] = None) -> Dict[str, List[str]]:
+    """Retrieve metrics for a given resource.
+
+    Args:
+      resource_id (str): The resource ID for which to lookup the metric.
+      metrics (str): A comma separated list of metrics to retrieve. E.g.
+          'Percentage CPU,Network In'.
+      from_date (datetime.datetime): Optional. A start date from which to get
+          the metric. If passed, toDate is also required.
+      to_date (datetime.datetime): Optional. An end date until which to get the
+          metric. If passed, fromDate is also required.
+      interval (str): An interval for the metrics, e.g. 'PT1H' will output
```

Comment:
Is this defined somewhere?

Reply:
There are some values in https://docs.microsoft.com/en-us/azure/azure-monitor/platform/rest-api-walkthrough but I haven't found any actual documentation talking about this